hpx/parallel/segmented_algorithms/detail/transfer.hpp

```diff
@@ -259,17 +259,18 @@ namespace hpx { namespace parallel { inline namespace v1
 
         ///////////////////////////////////////////////////////////////////////
         // segmented implementation
-        template <typename Algo, typename ExPolicy, typename InIter, typename OutIter>
+        template <typename Algo, typename ExPolicy, typename FwdIter,
+            typename OutIter>
         typename util::detail::algorithm_result<
-            ExPolicy, std::pair<InIter, OutIter>
+            ExPolicy, std::pair<FwdIter, OutIter>
         >::type
-        transfer_(ExPolicy&& policy, InIter first, InIter last, OutIter dest,
+        transfer_(ExPolicy&& policy, FwdIter first, FwdIter last, OutIter dest,
```

Comment:
Is it okay to use ```OutIter``` as naming in here?
This review is applied in other places in this file.